src/huggingface_hub/inference/_providers/__init__.py

```diff
@@ -118,21 +119,30 @@
 }
 
 
-def get_provider_helper(provider: PROVIDER_T, task: str) -> TaskProviderHelper:
+def get_provider_helper(provider: PROVIDER_T, task: str, model: str) -> TaskProviderHelper:
     """Get provider helper instance by name and task.
 
     Args:
         provider (str): Name of the provider
         task (str): Name of the task
-
+        model (str): Name of the model
     Returns:
         TaskProviderHelper: Helper instance for the specified provider and task
 
     Raises:
         ValueError: If provider or task is not supported
     """
-    if provider not in PROVIDERS:
-        raise ValueError(f"Provider '{provider}' not supported. Available providers: {list(PROVIDERS.keys())}")
+    if provider != "auto" and provider not in PROVIDERS:
+        raise ValueError(
+            f"Provider '{provider}' not supported. Available providers: {['auto'] + list(PROVIDERS.keys())}, "
+            "'auto' will automatically select the first provider available for the model, sorted "
+            "by the user's order in https://hf.co/settings/inference-providers."
+        )
+
+    if provider == "auto":
+        provider_mapping = _fetch_inference_provider_mapping(model)
+        provider = next(iter(provider_mapping))
+
```

Comment:
```suggestion
    if provider == "auto":
        provider_mapping = _fetch_inference_provider_mapping(model)
        provider = next(iter(provider_mapping))

    if provider not in PROVIDERS:
        raise ValueError(
            f"Provider '{provider}' not supported. Available values: 'auto' or any provider from {list(PROVIDERS.keys())}."
            "Passing 'auto' (default value) will automatically select the first provider available for the model, sorted "
            "by the user's order in https://hf.co/settings/inference-providers."
        )
```

better this way IMO to decorrelate the "policy" from the "is this provider supported" check. Kinda corner case but it's possible that a provider is listed on the Hub but not yet supported by the Python client.